### DISCUSSION TITLE
How to create custom shape

### DISCUSSION THREAD

**Discussion by MANOUKIAN25** [InteractiveComputerGraphics/PositionBasedDynamics#150] [Q&A]
Hi, 

From the function "addRegularTriangleModel" from the "simulationModel.cpp" file.

The function creates a rectangle shape according to the height and width given, so I wanted to know please if you could provide an example on how to create different shapes like "trapezoid, isosceles trapezoid, kite" and eventually a combination of these simple shapes together to generate a unique shape .

If it is possible please I'd like also to know how the function "addRegularTriangleModel" is implemented and what each variables refer to (if you have any paper I could look at).

Amazing lib by the way !

Thank you and have a nice week end.

**Comment by janbender** [comment]
Just take a look at the implementation if you want to know how the function is implemented: 
https://github.com/InteractiveComputerGraphics/PositionBasedDynamics/blob/master/Simulation/SimulationModel.cpp#L614

If you need different shapes, the easiest way is to load an OBJ file.
